src/factors/DynPoint2D.jl

```diff
@@ -24,21 +24,16 @@ mutable struct DynPoint2DynPoint2{T <: SamplableBelief} <: AbstractRelativeRoots
 end
 
 getSample(cfo::CalcFactor{<:DynPoint2DynPoint2}, N::Int=1) = (rand(cfo.factor.z,N), )
-function (cfo::CalcFactor{<:DynPoint2DynPoint2})(
-            res::AbstractVector{<:Real},
-            z,
-            xi,
-            xj  )
+
+function (cfo::CalcFactor{<:DynPoint2DynPoint2})(z, xi, xj)
   #
   dt = Dates.value(cfo.metadata.fullvariables[2].nstime - cfo.metadata.fullvariables[1].nstime)*1e-9   # roughly the intended use of userdata
-  res[1:2] = z[1:2] - (xj[1:2] - (xi[1:2]+dt*xi[3:4]))
-  res[3:4] = z[3:4] - (xj[3:4] - xi[3:4])
-  nothing
+  res12 = z[1:2] - (xj[1:2] - (xi[1:2]+dt*xi[3:4]))
+  res34 = z[3:4] - (xj[3:4] - xi[3:4])
+  return [res12; res34]
```

Comment:
we'll see how the allocations go.  if bad then we can turn to using the cfo.residual container -- likely best done after consolidating CPT?

Reply:
The function will probably get inlined, so it shouldn't make much of a difference. NLSolve reuses the residual in any case.